src/Sulu/Bundle/DocumentManagerBundle/Document/Subscriber/SecuritySubscriber.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\DocumentManagerBundle\Document\Subscriber;
+
+use Sulu\Component\DocumentManager\Event\ConfigureOptionsEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Responsible for integrating the security part of Sulu to the DocumentManager.
+ */
+class SecuritySubscriber implements EventSubscriberInterface
```

Comment:
This subscriber makes detecting the user more reusable. It's always set as an option, so that it can be reused in all the different events.